examples/linux-offsets.c

```diff
@@ -0,0 +1,259 @@
+        {"name", required_argument, NULL, 'n'},
+        {"domid", required_argument, NULL, 'd'},
+        {"json-kernel", required_argument, NULL, 'r'},
+        {"only-kpgd", no_argument, NULL, 'k'},
+    };
+
+    while ((c = getopt_long (argc, argv, "n:d:kr:", long_opts, &long_index)) != -1)
+        switch (c) {
+            case 'n':
+                domain = (void *)optarg;
+                init_flags |= VMI_INIT_DOMAINNAME;
+                break;
+            case 'd':
+                domid = strtoull(optarg, NULL, 0);
+                domain = (void *)&domid;
+                init_flags |= VMI_INIT_DOMAINID;
+                break;
+            case 'k':
+                only_output_kpgd = 1;
+                break;
+            case 'r':
+                kernel_profile = optarg;
+                break;
+            default:
+                show_usage(argv[0]);
+                return 1;
+        }
+
+    if (optind != argc) {
+        fprintf(stderr, "Unrecognized argument: %s\n", argv[optind]);
+        show_usage(argv[0]);
+        return 1;
+    }
+
+    if (!domain) {
+        fprintf(stderr, "You have to specify --name or --domid!\n");
+        show_usage(argv[0]);
+        return 1;
+    }
+
+    if ((init_flags & VMI_INIT_DOMAINNAME) && (init_flags & VMI_INIT_DOMAINID)) {
+        fprintf(stderr, "Both domain ID and domain name provided!\n");
+        show_usage(argv[0]);
+        return 1;
+    }
+
+    if (!kernel_profile) {
+        fprintf(stderr, "You have to specify path to kernel JSON profile!\n");
+        show_usage(argv[0]);
+        return 1;
+    }
+
+    if (VMI_FAILURE == vmi_get_access_mode(vmi, domain, init_flags, NULL, &mode)) {
+        printf("Failed to get access mode\n");
+        goto done;
+    }
+
+    /* initialize the libvmi library */
+    if (VMI_FAILURE == vmi_init(&vmi, mode, domain, init_flags | VMI_INIT_EVENTS, NULL, NULL)) {
+        fprintf(stderr, "Failed to init LibVMI library.\n");
+        goto done;
+    }
+
+    signal(SIGINT, sigint_handler);
+
+    config = g_hash_table_new_full(g_str_hash, g_str_equal, g_free, g_free);
+    if (!config) {
+        fprintf(stderr, "Failed to create GHashTable!\n");
+        goto done;
+    }
+
+    g_hash_table_insert(config, g_strdup("os_type"), g_strdup("Linux"));
+    g_hash_table_insert(config, g_strdup("rekall_profile"), g_strdup(kernel_profile));
+
+    if (VMI_PM_UNKNOWN == vmi_init_paging(vmi, VMI_PM_INITFLAG_TRANSITION_PAGES) ) {
+        fprintf(stderr, "Failed to init LibVMI paging.\n");
+        goto done;
+    }
+
+    os_t os = vmi_init_profile(vmi, VMI_CONFIG_GHASHTABLE, config);
+    if (VMI_OS_LINUX != os) {
+        fprintf(stderr, "OS is not Linux\n");
+        goto done;
+    }
+
+    g_hash_table_remove(config, "rekall_profile");
+
+    if ( vmi_are_events_pending(vmi) > 0 )
+        vmi_events_listen(vmi, 0);
+
+    // the vm is already paused if we've got here
+    os = vmi_init_os(vmi, VMI_CONFIG_GHASHTABLE, config, NULL);
```

Comment:
You could just use `VMI_CONFIG_JSON_PATH` and pass the json path to LibVMI directly, no need to construct the ghashtable just for the json path.

Reply:
Incorporated